INSVCTwoFluidStaggeredHierarchyIntegrator.cpp

```diff
@@ -456,48 +459,48 @@ INSVCTwoFluidStaggeredHierarchyIntegrator::integrateHierarchy(const double curre
         ghost_cell_fill.initializeOperatorState(ghost_cell_comp, d_hierarchy, 0, d_hierarchy->getFinestLevelNumber());
         ghost_cell_fill.fillData(0.0);
     }
-
-    // set-up RHS to treat viscosity and drag with backward Euler: f(n) + C*theta_i(n)*u_i(n) for  i = n, s
-    for (int ln = 0; ln <= d_hierarchy->getFinestLevelNumber(); ++ln)
-    {
-        Pointer<PatchLevel<NDIM>> level = d_hierarchy->getPatchLevel(ln);
-        for (PatchLevel<NDIM>::Iterator p(level); p; p++)
-        {
-            Pointer<Patch<NDIM>> patch = level->getPatch(p());
-            Pointer<CartesianPatchGeometry<NDIM>> pgeom = patch->getPatchGeometry();
-            Pointer<SideData<NDIM, double>> un_data = patch->getPatchData(un_sc_idx);
-            Pointer<SideData<NDIM, double>> us_data = patch->getPatchData(us_sc_idx);
-            Pointer<SideData<NDIM, double>> f_un_data = patch->getPatchData(f_un_sc_idx);
-            Pointer<SideData<NDIM, double>> f_us_data = patch->getPatchData(f_us_sc_idx);
-            Pointer<CellData<NDIM, double>> thn_data = patch->getPatchData(thn_cc_idx);
-            IntVector<NDIM> xp(1, 0), yp(0, 1);
-
-            for (SideIterator<NDIM> si(patch->getBox(), 0); si; si++) // side-centers in x-dir
-            {
-                const SideIndex<NDIM>& idx = si();                                         // axis = 0, (i-1/2,j)
-                CellIndex<NDIM> idx_c_low = idx.toCell(0);                                 // (i-1,j)
-                CellIndex<NDIM> idx_c_up = idx.toCell(1);                                  // (i,j)
-                double thn_lower = 0.5 * ((*thn_data)(idx_c_low) + (*thn_data)(idx_c_up)); // thn(i-1/2,j)
-                (*f_un_data)(idx) = (*f_un_data)(idx) + C * thn_lower * (*un_data)(idx);
-                (*f_us_data)(idx) = (*f_us_data)(idx) + C * convertToThs(thn_lower) * (*us_data)(idx);
-            }
-
-            for (SideIterator<NDIM> si(patch->getBox(), 1); si; si++) // side-centers in y-dir
-            {
-                const SideIndex<NDIM>& idx = si();                                         // axis = 1, (i,j-1/2)
-                CellIndex<NDIM> idx_c_low = idx.toCell(0);                                 // (i,j-1)
-                CellIndex<NDIM> idx_c_up = idx.toCell(1);                                  // (i,j)
-                double thn_lower = 0.5 * ((*thn_data)(idx_c_low) + (*thn_data)(idx_c_up)); // thn(i,j-1/2)
-                (*f_un_data)(idx) = (*f_un_data)(idx) + C * thn_lower * (*un_data)(idx);
-                (*f_us_data)(idx) = (*f_us_data)(idx) + C * convertToThs(thn_lower) * (*us_data)(idx);
-            }
-        } // patches
-    }     // levels
+    // set-up RHS to treat viscosity and drag with backward Euler or Implicit Trapezoidal Rule: 
+    // RHS = f(n) + C*theta_i(n)*u_i(n) + D1*(pressure + viscous + drag) for  i = n, s
+    const double D1;
+    const double D2;
```

Comment:
These should not be `const`.
```suggestion
    double D1;
    double D2;
```